### PR TITLE
Add brand field to app manifest

### PR DESCRIPTION
**Pull request by przlada**
I want to merge this change because it adds brand information to the app manifest and app model. For now, it consists only of the default image logo, but it could be easily extended with additional information in the future. I would like to use this PR as a base to discuss the following:

1. New optional field in the App manifest: `brand`
    ```
    brand: {
      logo: {
        default: string // url
      },
    }
    ```

2. Icon image validation steps:
    1. Fetch the image from `brand.logo.default` URL
    2. Check and accept only PNG image format as input
    3. Check if the image has the minimum required size (256px) and is square

3. Thumbnails generation of icon images is possible only to formats `WebP` and `PNG` (lossless compression)
`
4. Mutation `appFetchManifest` fetches the logo image, validates it, and returns its thumbnail on the fly as a string in the form of base64 encoded data URL (for example: `data:image/webp;base64,UklGRtoPAABXRUJQV...`). A new optional field `brand` of type `AppManifestBrand` is added to the `Manifest` object type:
```graphql
enum IconThumbnailFormatEnum {
  ORIGINAL
  WEBP
}

type AppManifestBrand @doc(category: "Apps") {
  logo: AppManifestBrandLogo!
}

type AppManifestBrandLogo @doc(category: "Apps") {
  default(size: Int format: IconThumbnailFormatEnum = ORIGINAL): String!
}
```
Brand data can be used by the dashboard for displaying Manifest preview: 
manifest_preview

6. Mutation `appInstall` fetches the logo image, validates it, and stores it in the Saleor's file storage. `ImageField` is added to `App` and `AppInstallation` models, and `App` and `AppInstallation` graphQL object types have a new optional field `brand` of type `AppBrand`. Thanks to that dashboard can display apps and app installations icons with logos. 
```graphql
"""Represents the app's brand data."""
type AppBrand @doc(category: "Apps") {
  logo: AppBrandLogo!
}

type AppBrandLogo @doc(category: "Apps") {
  """URL to the default logo image."""
  default(size: Int format: IconThumbnailFormatEnum = ORIGINAL): String!
}
```

### TODO
* [x] Add `uuid` field to `App` and `AppInstallation` models
* [x] Handle thumbnail generation for `App` and `AppInstallation`
* [x] Move and reformat code to avoid duplication (image validation, etc.)
* [x] Add tests

# Impact

* [x] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [x] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/app/types.py [hunk 1]
```diff
@@ -221,6 +236,57 @@ class Meta:
...
+    default = IconThumbnailField(
+        graphene.String, required=True, description="URL to the default logo image."
+    )
+
+    class Meta:
+        doc_category = DOC_CATEGORY_APPS
+
+
+class AppBrand(BaseObjectType):
+    logo = graphene.Field(AppBrandLogo, required=True)
```
Comment (IKarbowiak) [inline saleor/graphql/app/types.py:279]: Lack of description 👮‍♀️

File: saleor/graphql/core/enums.py [hunk 2]
```diff
@@ -92,6 +92,8 @@ def to_enum(enum_cls, *, type_name=None, **options) -> graphene.Enum:
 PermissionEnum = graphene.Enum("PermissionEnum", get_permissions_enum_list())
 TimePeriodTypeEnum = to_enum(TimePeriodType)
 ThumbnailFormatEnum = to_enum(ThumbnailFormat)
+IconThumbnailFormatEnum = to_enum(IconThumbnailFormat)
```
Comment (IKarbowiak) [inline saleor/graphql/core/enums.py:95]: Do you think that `ThumbnailFormat` and `IconThumbnailFormat` will differ in the feature? I;m wondering if we cannot use the same enum?

File: saleor/thumbnail/utils.py [hunk 3]
```diff
@@ -220,4 +237,66 @@ def process_image(self, image, save_kwargs):
...
+
+class ProcessedIconImage(ProcessedImage):
+    LOSSLESS_WEBP = True
+
+
+def is_icon_image_mimetype(mimetype: Optional[str]) -> bool:
+    return mimetype in ICON_MIME_TYPES
+
+
+def get_filename_from_url(url: str) -> str:
```
Comment (IKarbowiak) [inline saleor/thumbnail/utils.py:251]: Could we add a docstring here?

**Review by IKarbowiak — APPROVED** [review]

File: saleor/app/manifest_validations.py [hunk 4]
```diff
@@ -349,3 +360,30 @@ def clean_author(author) -> Optional[str]:
     raise ValidationError(
         "Incorrect value for field: author", code=AppErrorCode.INVALID.value
     )
+
+
+def fetch_icon_image(url: str, field_name: str) -> ContentFile:
+    filename = get_filename_from_url(url)
+    try:
+        with requests.get(url, stream=True, timeout=REQUEST_TIMEOUT) as res:
```
Comment (maarcingebala) [inline saleor/app/manifest_validations.py:368]: What do you think about fetching this image async in a task with some retries if the initial request doesn't work? I'm just thinking that maybe the logo can be loaded later and doesn't have to slow down the app installation process. I won't argue about that; it's just an idea.

Comment (korycins) [reply saleor/app/manifest_validations.py:368]: Yeah, I am still thinking that logo is something optional, and if we can't fetch it, we should just allow to install the app and maybe try fetching it once again.

**Review by maarcingebala — APPROVED** [review]

**Comment by maarcingebala** [comment]
@przlada The failing test is fixed in https://github.com/saleor/saleor/pull/12964.

**Review by szczecha — APPROVED** [review]

**Pull request merged**